Portal/View/SingleProperty.swift

```diff
@@ -0,0 +1,13 @@
+//
+//  SingleProperties.swift
+//  Portal
+//
+//  Created by Argentino Ducret on 8/29/17.
+//  Copyright © 2017 Guido Marucci Blas. All rights reserved.
+//
+
+struct SingleProperty<T> {
```

Comment:
I think it would be better to have

```swift
enum PropertyChange<T> {

  case change(to: T)
  case noChange

}
```

and update `TextViewChageSet`'s `textType` property to be of type `PropertyChange`.